src/main/java/com/google/devtools/build/lib/bazel/bzlmod/Selection.java

```diff
@@ -165,16 +170,100 @@ private static SelectionGroup computeSelectionGroup(
         allowedVersionSet.ceiling(module.getVersion()));
   }
 
+  private static ImmutableMap<UnresolvedModuleKey, ImmutableList<UnresolvedModule>>
+      computeCompatibilityLevelRangeDeps(
```

Comment:
could you please add some comments around? I understand that you wanted early review comments, but the code in this file is very dense and it's taking me a lot of time to get through. Some brief comments describing the intent would go a long way in speeding up the reviewing process.

Reply:
I added some comments. Most of them in this function, as I had a lot elsewhere.